src/app/pages/portal/portal.component.ts

```diff
@@ -408,4 +427,89 @@ export class PortalComponent implements OnInit, OnDestroy {
     });
   }
 
+  private updateSearchLayers(result: SearchResult) {
+    if (result === undefined) {
+      this.clearAllSearchLayers();
+      return;
+    }
+
+    if (this.contextState.context$.value === undefined ||
+      result === undefined ) { return; }
+
+    this.searchLayers = (this.contextState.context$.value as any).searchLayers;
+    const  searchType = (result.source.constructor as typeof SearchSource).type;
+    const layers = this.searchLayers[searchType];
+
+    layers.forEach((layerInfo: string | LayerOptions | Layer) => {
+      if (layerInfo instanceof Layer) {
+        if (this.searchRelatedAddedLayers.has(searchType)) {
```

Comment:
Make a method named addSearchLayer that'll receive a layer and a searchType and do that. Use it below as well.